app/javascript/components/patient/laboratory/Laboratory.js

```diff
@@ -44,29 +36,92 @@ class Laboratory extends React.Component {
     });
   };
 
+  toggleDeleteModal = () => {
+    let current = this.state.showDeleteModal;
+    this.setState({
+      showDeleteModal: !current,
+      delete_reason: null,
+      delete_reason_text: null,
+    });
+  };
+
+  handleChange = event => {
+    this.setState({ [event.target.id]: event.target.value });
+  };
+
+  handleDeleteSubmit = () => {
+    let deleteReason = this.state.delete_reason;
+    if (deleteReason === 'Other' && this.state.delete_reason_text) {
+      deleteReason += ', ' + this.state.delete_reason_text;
+    }
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .delete(window.BASE_PATH + '/laboratories/' + this.props.lab.id, {
+          data: {
+            patient_id: this.props.patient.id,
+            delete_reason: deleteReason,
+          },
+        })
+        .then(() => {
+          this.setState({ loading: false, showDeleteModal: false });
+          location.reload(true);
+        })
+        .catch(error => {
+          this.setState({ loading: false, showDeleteModal: false });
+          reportError(error);
+        });
+    });
+  };
+
   render() {
     return (
       <React.Fragment>
         {!this.props.lab.id && (
-          <Button onClick={() => this.setState({ showModal: true, loading: false })}>
+          <Button onClick={() => this.setState({ showEditModal: true, loading: false })}>
```

Comment:
this click function is used in 3 places, can you break it into a toggle method like we do with delete?

Reply:
Great suggestion. Done.